src/python/tests/genome_metadata/test_dump.py

```diff
@@ -182,13 +302,129 @@ def test_get_genome_metadata(
 
     Args:
         mock_session: A mock of `sqlalchemy.orm.Session()` class.
+        db_name:
         meta_data: `meta` table content in a list of named tuples.
         output: Expected genome metadata dictionary.
         expectation: Context manager for the expected exception (if any).
     """
+    # pylint: disable=too-many-positional-arguments
     mock_result.unique.return_value = mock_result
     mock_result.all.return_value = meta_data
     mock_session.execute.return_value = mock_result
     with expectation:
-        result = dump.get_genome_metadata(mock_session)
+        result = dump.get_genome_metadata(mock_session, db_name)
         assert not DeepDiff(result, output)
+
+
+@pytest.mark.parametrize(
+    "arg_list, expected",
+    [
+        param(
+            ["--host", "localhost", "--port", "42", "--user", "me", "--database", "test_db"],
+            {
+                "host": "localhost",
+                "port": 42,
+                "user": "me",
+                "password": None,
+                "url": make_url("mysql://me@localhost:42/test_db"),
+                "database": "test_db",
+                "metafilter": None,
+                "meta_update": False,
+                "append_db": False,
+                "log_file": None,
+                "log_level": "WARNING",
+                "log_file_level": "DEBUG",
+            },
+            id="Default args",
+        ),
+        param(
+            [
+                "--host",
+                "localhost",
+                "--port",
+                "42",
+                "--user",
+                "me",
+                "--database",
+                "test_db",
+                "--metafilter",
+                f"{__file__}",
+                "--append_db",
+            ],
+            {
+                "host": "localhost",
+                "port": 42,
+                "user": "me",
+                "password": None,
+                "url": make_url("mysql://me@localhost:42/test_db"),
+                "database": "test_db",
+                "metafilter": __file__,
+                "meta_update": False,
+                "append_db": True,
+                "log_file": None,
+                "log_level": "WARNING",
+                "log_file_level": "DEBUG",
+            },
+            id="Filter, non-default args",
+        ),
+    ],
+)
+def test_parse_args(arg_list: list[str], expected: dict) -> None:
+    """Tests the `dump.parse_args()` function."""
+    # pylint: disable=too-many-positional-arguments
+    args = dump.parse_args(arg_list)
+    if args.metafilter:
+        # DeepDiff is not able to compare two objects of Path type, so convert it to string
+        setattr(args, "metafilter", str(args.metafilter))
+    assert not DeepDiff(vars(args), expected)
+
+
+@pytest.mark.parametrize(
+    "arg_list, db_url, metafilter, meta_update, append_db, stdout",
```

Comment:
```suggestion
    ("arg_list", "db_url", "filter_file", "meta_update", "append_db", "stdout"),
```
It seems `pytest` has recently updated its code to allow this, much more readable, approach.